opaque_keys/edx/locator.py

```diff
@@ -260,7 +267,45 @@ def _from_string(cls, serialized):
         if parse['version_guid']:
             parse['version_guid'] = cls.as_object_id(parse['version_guid'])
 
-        return cls(**{key: parse.get(key) for key in cls.KEY_FIELDS})
+        parsed_values = tuple(parse.get(key) for key in cls.KEY_FIELDS)
+
+        # Include the class name in the cache key so that we don't have
+        # collisions with sub-classes.
+        cache_key = (cls.__class__, parsed_values)
```

Comment:
The comment says class name, the code uses the class object.  The latter is probably more robust (could have multiple classes with the same name in different packages), so just update the comment?